plugins/include/open62541/plugin/accesscontrol_default.h

```diff
@@ -18,6 +18,11 @@ typedef struct {
     UA_String password;
 } UA_UsernamePasswordLogin;
 
+typedef UA_StatusCode (*UA_UsernamePasswordLoginCallback)
+    (const UA_String *userName, const UA_ByteString *password,
+    size_t usernamePasswordLoginSize, const UA_UsernamePasswordLogin
+    *usernamePasswordLogin, void *loginContext);
```

Comment:
The login context could be a double-pointer.
So the context can be set (replaced) from within the login context.

Reply:
Most contexts are simple void* which are easy to use.  I only encountered double void pointers in UA_GlobalNodeLifecycle and for me this feature makes correct memory management hard.  For my use case, single pointer is enough and standard.  Double pointer feels over engineered, but this impression could change when I would need it.

Reply:
The activateSession API for the plugin is as follows:
```
UA_StatusCode
activateSession(UA_Server *server, UA_AccessControl *ac,
                        const UA_EndpointDescription *endpointDescription,
                        const UA_ByteString *secureChannelRemoteCertificate,
                        const UA_NodeId *sessionId,
                        const UA_ExtensionObject *userIdentityToken,
                        void **sessionContext);
```
The idea is that activateSession adds a context pointer to the session.
I think many users will want to attach the context pointer in the place where the username is evaluated.
So please keep the double-pointer.

Reply:
To pass a void pointer to a callback is common pattern in C.  The usual case, where the callback gets some memory to operate on or link to external data structures, is covered.  It is uncommon that some context is needed or globally changed during username and password validation.  But it is still possible with a single void pointer.  Just install a pointer to a struct where the callback has a field which it can change.  A single void pointer makes common things easy and complicated things possible.  So I think single pointer is a good API in this case.

Reply:
We should be consistent with our overall API.
This is how the AccessControl plugin API works:
https://github.com/open62541/open62541/blob/bf3b5945da2d744706d626fdf8ca42b39e5ee7ed/include/open62541/plugin/accesscontrol.h#L47

We need the double-pointer, because we want to set a void-pointer that is stored in the session.
But we don't define the session struct publicly.

I think users who use your username callback will want to do the same: Attach some custom data to the session.

The `sessionContext` double-ptr is in addition to the login context.
I would prefer that you forward the server-struct as a pointer instead of the login context.
If the login context is stored in the accesscontrol plugin, then we can get it via the server pointer anyway.